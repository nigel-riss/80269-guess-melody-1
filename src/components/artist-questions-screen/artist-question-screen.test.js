import React from 'react';
import renderer from 'react-test-renderer';

import ArtistQuestionsScreen from './artist-questions-screen.jsx';

const mock = {
  question: {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
    },
    answers: [
      {
        picture: `vinyl.png`,
        artist: `John Snow`,
      },
      {
        picture: `vinyl.png`,
        artist: `Jack Daniels`,
      },
      {
        picture: `vinyl.png`,
        artist: `Jim Beam`,
      },
    ],
  },
};

it(`ArtistQuestionsScreen is rendered correctly`, () => {
  const {question} = mock;

  const tree = renderer.create(<ArtistQuestionsScreen
    onAnswer={jest.fn()}
    question={question}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
