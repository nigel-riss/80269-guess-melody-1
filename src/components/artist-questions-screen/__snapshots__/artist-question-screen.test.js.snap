// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionsScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      type="button"
    />
    <audio />
  </div>
  <form
    className="game__artist"
    onChange={[MockFunction]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="./img/vinyl.png"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="./img/vinyl.png"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="./img/vinyl.png"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
